src/main/java/net/sf/jabref/external/push/PushToApplicationButton.java

```diff
@@ -130,22 +126,26 @@ private void buildPopupMenu() {
 
     /**
      * Update the PushButton to default to the given application.
+     *
      * @param i The List index of the application to default to.
      */
     private void setSelected(int i) {
-        this.selected = i;
+        selected = i;
         PushToApplication toApp = pushActions.get(i);
         pushButton.setIcon(toApp.getIcon());
         pushButton.setToolTipText(toApp.getTooltip());
         pushButton.setPreferredSize(buttonDim);
 
-        Globals.prefs.put("pushToApplication", toApp.getName());
+        // Store the last used application
+        Globals.prefs.put(Globals.prefs.PUSH_TO_APPLICATION, toApp.getApplicationName());
```

Comment:
Please use JabRefPreferences.PUSH_TO_APPLICATION instead. As we do not want to access static variables on instances. Theoretically, Eclipse should issue a warning for these kinds of things.